src/main/java/com/google/jenkins/plugins/computeengine/PreemptedCheckCallable.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jenkins.plugins.computeengine;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import hudson.model.TaskListener;
+import jenkins.security.MasterToSlaveCallable;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+
+/**
+ * Slave callback class checking if instance was preempted.
+ * All of code here is serialized and executed on node.
+ */
+final class PreemptedCheckCallable extends MasterToSlaveCallable<Boolean, IOException> {
+    private static final String METADATA_SERVER_URL = "http://metadata.google.internal/computeMetadata/v1/instance/preempted?wait_for_change=true";
+
+    private final TaskListener listener;
+
+    /**
+     * Callback constructor.
+     * @param listener Node listener on which we can add extra information from slave side.
+     */
+    PreemptedCheckCallable(TaskListener listener) {
+        this.listener = listener;
+    }
+
+    /**
+     * Actual callback code, executed on node side.
+     * Checks in Google metadata server if instance was preempted.
+     * 
+     * See https://cloud.google.com/compute/docs/instances/create-start-preemptible-instance#detecting_if_an_instance_was_preempted
+     * 
+     * @return True if node was preempted.
+     * @throws IOException Exception when calling Google metadata API
+     */
+    @Override
+    public Boolean call() throws IOException {
+        HttpTransport transport = new NetHttpTransport();
```

Comment:
This method's content should exist within a client abstraction similar to the compute client. Also, it's client should be created with the ClientFactory. This de-duplicates much of this logic, and ensures the user agent string for the plugin is preserved. Also, a generated java client lib for the API should be used if it's available.

Reply:
But we will never use this methods in any different places..
So we dont need extra (and rather dont want) client for it..
And the smaller its the faster it works, cause everything here must be serialised and send to slave node (cause its called from slave itself)

I didnt found any lib for that purpose.. I tried but i failed.. if you know any let me know..

Reply:
I checked with our libraries team, and there currently doesn't exist a client lib for this API. That said, it would still be preferable to maintain the consistency of client abstractions we've established throughout the code base, and move this API logic into its own client class. It allows for better testing and separation of concerns.